tests/explainers/test_linear.py

```diff
@@ -130,15 +122,15 @@ def test_perfect_colinear():
     assert np.abs(shap_values.sum(1) - model.predict(X) + model.predict(X).mean()).sum() < 1e-7
 
 def test_shape_values_linear_many_features():
-    Ridge = pytest.importorskip('sklearn.linear_model').Ridge
 
-    np.random.seed(0)
+    Ridge = pytest.importorskip('sklearn.linear_model').Ridge
 
     coef = np.array([1, 2]).T
 
+    rs = np.random.RandomState(0)
     # generate linear data
```

Comment:
@connortann Was this accidentally missed out or it was set to seed=0 explicitly? I was expecting we would use the `random_seed` fixture here.

(If it was set to 0 explicitly, then we'll need the comment here to refer back to #2960 as well)

Reply:
Ah yes, good spot. That was indeed a flaky test; I've updated the list in the issue accordingly.

Reply:
I've resisted the urge to do any debugging or fixing of the underlying failures in this PR, as the diff is already very large! Hoping we can address that after the next release.

Reply:
Just came back to this and realized this comment wasn't addressed (I suppose I wasn't clear in my comment). I'll raise a PR.

Reply:
Ah, apologies if I misunderstood you - I had updated the list of flaky tests in #2960, but not done any "fixing" of the tests other than pinning the seed here to something that passes.